### PR TITLE
[utils] update SuccessOrExit for error tracking

### PR DESCRIPTION
**Pull request by wgtdkp**
This commit updates the `SuccessOrExit` macro to accept an action before goto "exit". It allows us to add a log when error happens. For example, we can write:

```
SuccessOrExit((error = writeDatasetToSettings()), otbrLogErr("Failed to write dataset to settings"));
```

So that we know exactly which step a method fails at.

------

This commit also prints the error/status code for macro `SuccessOrDie`.

### CODE REVIEW

File: src/common/code_utils.hpp [hunk 1]
```diff
@@ -89,14 +90,14 @@
...
-        {                                                                    \
-            otbrLogEmerg("FAILED %s:%d - %s", __FILE__, __LINE__, aMessage); \
-            exit(-1);                                                        \
-        }                                                                    \
+#define SuccessOrDie(aStatus, aMessage)                                                   \
+    do                                                                                    \
+    {                                                                                     \
+        if ((aStatus) != 0)                                                               \
+        {                                                                                 \
+            otbrLogEmerg("FAILED %s:%d - %s: %d", __FILE__, __LINE__, aMessage, aStatus); \
```
Comment (zhanglongxia) [inline src/common/code_utils.hpp:98]: How about putting the status value in front of the message string? The length of the message string is uncertain. Putting the status value at the beginning makes the developer easier to find the status value.

Comment (wgtdkp) [reply src/common/code_utils.hpp]: SGTM, updated!

**Review by zhanglongxia — APPROVED** [review]
LGTM

**Review by bukepo — APPROVED** [review]

File: tests/scripts/openwrt [hunk 2]
```diff
@@ -43,7 +43,7 @@ readonly CONTAINER_NAME
 
 do_prepare()
 {
-    docker create --name "${CONTAINER_NAME}" --rm -v openwrt-bin:/home/build/openwrt/bin -it openwrt/sdk
+    docker create --name "${CONTAINER_NAME}" --rm -v openwrt-bin:/home/build/openwrt/bin -it openwrt/sdk:22.03.3
```
Comment (bukepo) [inline tests/scripts/openwrt:46]: Is this change intentional?

Comment (wgtdkp) [reply tests/scripts/openwrt:46]: Yes, current openwrt CI is broken because the openwrt/sdk:latest has incompatible changes that otbr can't build it the docker image (e.g. no `sudo`, no `/home/build/openwrt` directory). It's probably more stable to stick with the latest "good" build to unblock the CI on the main branch (for example, https://github.com/openthread/ot-br-posix/pull/1930).

I will ask the commits being merged separately.

**Comment by wgtdkp** [comment]
@jwhui When merging this PR, please merge the two commits separately. The second one is for fixing the broken openwrt CI tests

**Review by jwhui — APPROVED** [review]
